### PR TITLE
[DOC] Added paper citation for DeformConv2d

### PR DESCRIPTION
**Pull request by hassiahk**
As described in https://github.com/pytorch/vision/issues/2975, I have added citation of the paper in the documentation of DeformConv2d and also mentioned about DeformConv v2.

### CODE REVIEW

File: torchvision/ops/deform_conv.py [hunk 1]
```diff
@@ -19,8 +19,11 @@ def deform_conv2d(
     dilation: Tuple[int, int] = (1, 1),
     mask: Optional[Tensor] = None,
 ) -> Tensor:
-    """
-    Performs Deformable Convolution, described in Deformable Convolutional Networks
+    r"""
+    Performs Deformable Convolution v2, described in
```
Comment (datumbox) [inline torchvision/ops/deform_conv.py:23]: Lint is complaining about the training space in this line.

**Review by datumbox — COMMENTED** [review]
Thanks for the PR. Could you please resolve the lint problem?

**Comment by hassiahk** [comment]
I fixed it! Can you review it once again?

File: torchvision/ops/deform_conv.py (hunk 1 above)
Comment (hassiahk) [reply torchvision/ops/deform_conv.py]: Fixed it!

**Review by fmassa — COMMENTED** [review]
Thanks for the PR!

Can you also keep the citation from the previous paper? The current implementation in torchvision supports both modes, if `mask is None` we dispatch to the original deform conv paper.

**Comment by hassiahk** [comment]
I have added the citation of the original paper too. Kindly review it once again.

**Review by fmassa — APPROVED** [review]
Thanks a lot!
